File: pyround1/1539.第-k-个缺失的正整数.py
#
# @lc app=leetcode.cn id=1539 lang=python3
#
# [1539] 第 k 个缺失的正整数
#

# @lc code=start
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        count = 0
        nums = list(range(1, (arr[-1] + 1)))
        for num in nums:
            if num not in arr:
                count += 1
                if count == k:
                    ans = num
                    break 
        if count < k:
            ans = nums[-1]
            while count < k:
                ans += 1
                count += 1
        return ans
        
                

# @lc code=end

File: 1539.第-k-个缺失的正整数.cpp
/*
 * @lc app=leetcode.cn id=1539 lang=cpp
 *
 * [1539] 第 k 个缺失的正整数
 */

// @lc code=start
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        int left = 0, right = arr.size();
        while(left < right){
            int mid = (left + right) / 2;
            if(arr[mid] - mid >= k+1)
                right = mid;
            else
            {
                left = mid+1;
            }
            
        }
        return k + left;

    }
};
// @lc code=end

